websites/permissions.py

```diff
@@ -0,0 +1,165 @@
+""" permissions for websites """
+from django.contrib.auth.models import Group, Permission
+from django.db import transaction
+from guardian.shortcuts import assign_perm
+from rest_framework.generics import get_object_or_404
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+from websites import constants
+from websites.models import Website
+
+
+def assign_object_permissions(group_or_user, website, perms):
+    """
+    Assign appropriate permissions to a website admin group
+
+    Args:
+        group_or_user (Group or User): the group/user to add permissions to
+        website (Website): The website that the permissions are for
+        perms (list of str): The permissions to be assigned
+    """
+    for perm in perms:
+        try:
+            assign_perm(perm, group_or_user, website)
+        except Permission.DoesNotExist as err:
+            raise Permission.DoesNotExist(f"Permission '{perm}' not found") from err
+
+
+@transaction.atomic
+def create_website_groups(website):
+    """
+    Create groups and assign permissions for a website
+
+    Args:
+        website (Website): The website that the permissions are for
+    """
+    admin_group, _ = Group.objects.get_or_create(
+        name=f"{constants.ADMIN_GROUP}{website.uuid.hex}"
+    )
+    editor_group, _ = Group.objects.get_or_create(
+        name=f"{constants.EDITOR_GROUP}{website.uuid.hex}"
+    )
+    assign_object_permissions(admin_group, website, constants.PERMISSIONS_ADMIN)
+    assign_object_permissions(editor_group, website, constants.PERMISSIONS_EDITOR)
+
+
+@transaction.atomic
+def create_global_groups():
+    """Create the global groups for website permissions"""
+    admin_group, _ = Group.objects.get_or_create(name=constants.GLOBAL_ADMIN)
+    author_group, _ = Group.objects.get_or_create(name=constants.GLOBAL_AUTHOR)
+    for perm in constants.PERMISSIONS_ADMIN:
+        assign_perm(perm, admin_group)
+    assign_perm(constants.PERMISSION_ADD, admin_group)
+    assign_perm(constants.PERMISSION_ADD, author_group)
```

Comment:
General comment about this function, `create_website_groups`, and the management command: I think it would be helpful to return the number of groups created (maybe created _and_ updated) in these functions, and you could probably add some logic to exit out of them if the groups already exist with the right permissions. The management command could then report on the number of groups that were actually affected

Reply:
I went ahead and did this, but personally I feel the usefulness it provides is marginal, especially considering the extra code complexity it introduces and the expected low frequency (probably just once per environment) that this command will run.